models/eprop_synapse.h

```diff
@@ -0,0 +1,553 @@
+/*
+ *  eprop_synapse.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef EPROP_SYNAPSE_H
+#define EPROP_SYNAPSE_H
+
+// nestkernel
+#include "connection.h"
+#include "connector_base.h"
+#include "eprop_archiving_node.h"
+#include "target_identifier.h"
+#include "weight_optimizer.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: synapse, e-prop plasticity
+
+Short description
++++++++++++++++++
+
+Synapse type for e-prop plasticity
+
+Description
++++++++++++
+
+``eprop_synapse`` is an implementation of a connector model to create synapses between postsynaptic
+neurons :math:`j` and presynaptic neurons and :math:`i` for eligibility propagation (e-prop) plasticity.
+
+E-prop plasticity was originally introduced and implemented in TensorFlow in [1]_.
+
+The e-prop synapse triggers the calculation of the gradient at each spike
+over an interval that begins at the previous spike and ends at a cutoff specified by the user or the
+current spike, depending on which of the two time points is earlier.
+The gradient calculation is specific to the post-synaptic neuron and thus defined there.
+
+Eventually, it optimizes the weight with the specified optimizer.
+
+E-prop synapses require archiving of continuous quantities. Therefore e-prop
+synapses can only be connected to neuron models that are capable of
+archiving. So far, compatible models are ``eprop_iaf``,
+``eprop_iaf_adapt``, and ``eprop_readout``.
```

Comment:
From the sine-waves example, it seems to me that the synapse also works with `eprop_iaf_psc_delta`.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/3207/commits/f8bc10a607f8414f18b3bae2a780edbff186307a